src/jemalloc.c

```diff
@@ -2384,26 +2384,19 @@ imalloc_sample(static_opts_t *sopts, dynamic_opts_t *dopts, tsd_t *tsd,
     size_t usize, szind_t ind) {
 	void *ret;
 
-	/*
-	 * For small allocations, sampling bumps the usize.  If so, we allocate
-	 * from the ind_large bucket.
-	 */
-	szind_t ind_large;
-
 	dopts->alignment = prof_sample_align(dopts->alignment);
 	if (usize <= SC_SMALL_MAXCLASS) {
-		assert(((dopts->alignment == 0) ?
-		    sz_s2u(SC_LARGE_MINCLASS) :
-		    sz_sa2u(SC_LARGE_MINCLASS, dopts->alignment))
-			== SC_LARGE_MINCLASS);
-		ind_large = sz_size2index(SC_LARGE_MINCLASS);
-		size_t bumped_usize = sz_s2u(SC_LARGE_MINCLASS);
-		ret = imalloc_no_sample(sopts, dopts, tsd, bumped_usize,
-		    bumped_usize, ind_large);
+		size_t bumped_usize = PAGE_CEILING(usize);
```

Comment:
High-level question that applies to most of these function calls: What is the convention in Jemalloc for when and where you should apply alignment to a size? I have *not* applied it here (e.g. with `sz_sa2u` or similar) since the `arena_large_palloc` accepts the alignment as an argument. Is this correct?

Reply:
For the "normal" / non-sampled case, whenever there's a `usize` (usable size), it takes the alignment into the size class consideration already (in case the alignment would cause the size class to change, e.g. alloc 5k but asking for 4k aligned, resulting the 8k size class being used). So the convention is translating size to usize once in `imalloc_body`, and then pass the alignment argument downstream.

Here it's a special case, because we are adding another alignment and size bumping (on top of the already computed usize and alignment). The PAGE_CEILING is very close to sa2u, except the case where alignment is 8K (asked by user input), where sa2u(9k, 8k) == 16K. So we would want to change page ceiling to `sa2u(usize, dopts->alignment)` instead.